cdk/setups/analytics_setup.py

```diff
@@ -0,0 +1,60 @@
+
+from aws_cdk import core, aws_lambda as _lambda, aws_iam, aws_apigateway as apigateway
+
+def setup_analytics(stack: core.Stack):
+    # IAM Role for Lambda
+    lambda_role = aws_iam.Role(
+        stack, 'DashboardGeneratorRole',
+        assumed_by=aws_iam.ServicePrincipal('lambda.amazonaws.com'),
+        managed_policies=[
+            aws_iam.ManagedPolicy.from_aws_managed_policy_name('AWSLambda_FullAccess'),
+        ]
+    )
+    lambda_role.add_to_policy(aws_iam.PolicyStatement(
+        effect=aws_iam.Effect.ALLOW,
+        actions=[
+            'quicksight:DescribeDashboard',
+            'quicksight:GetDashboardEmbedUrl',
+            'quicksight:GetAuthCode'
+        ],
+        resources=['*'],
+    ))
+
+    # Lambda definition
+    lambda_dashboard_generator = _lambda.Function(
+        stack,
+        "QuickSightDashboardGenerator",
+        runtime=_lambda.Runtime.PYTHON_3_8,
+        handler="dashboard_generator.lambda_handler",
+        code=_lambda.Code.from_asset('visit/lambda_code/quicksight'),
+        role=lambda_role,
+        environment={

```

Comment:
> Note: Quicksight User ARN will be updated. 

Assuming this is what you meant by that. 
Pretty sure these values can be pulled from the CDK, assuming they are resources managed by our stack

Reply:
I'll look into finding a way to obtain the quicksight user arn dynamically. When it comes to the Quicksight User ARN, it needs to belong to a user role that has admin access to the makerspace quicksight dashboard. I currently only have read access to the clemson makerspace prod account, so I can't view the arn's of the user roles available.

Reply:
For the aws_account_id, I changed it to use core.Aws.ACCOUNT_ID to obtain it.